mirascope/core/mistral/_utils/_setup_call.py

```diff
@@ -74,18 +120,9 @@ def setup_call(
         client = (
             MistralAsyncClient() if inspect.iscoroutinefunction(fn) else MistralClient()
         )
-
-    def create_or_stream(
-        stream: bool,
-        **kwargs: Any,  # noqa: ANN401
-    ) -> (
-        ChatCompletionResponse
-        | Coroutine[Any, Any, ChatCompletionResponse]
-        | AsyncGenerator[ChatCompletionStreamResponse, None]
-        | Iterable[ChatCompletionStreamResponse]
-    ):
-        if stream:
-            return client.chat_stream(**kwargs)
-        return client.chat(**kwargs)
-
+    if isinstance(client, MistralAsyncClient):
```

Comment:
this whole thing with dataclasses and extra methods feels like overkill. I feel like we could just safely cast types here and call it a day? same for cohere

Reply:
Originally, I was going to use namedtuple, but there was a problem that namedtuple and generics could not be used together in 3.9.
I will remove dataclasses.